test/powershell/Modules/Microsoft.PowerShell.Utility/FormatHex.Tests.ps1

```diff
@@ -5,219 +5,460 @@
+                Path = @($inputFile1, $inputFile3)
+                Count = 2
+                Expected = $inputText1
+                secondExpected = $inputText3
+            }
+        )
+
+        foreach ($testCase in $testCases)
+        {
+            RunPathAndLiteralPathParameterTestCase $testCase
+        }
     }
 
-    # This test is to validate that integers can be piped to the format-hex
-    # and properly represented as characters in the string
-    It "ValidateIntegerInputThatPresentAsCharacters" {
+    Context "Encoding Parameter" {
+        $testCases = @(
+            @{
+                Name = "Can process ASCII encoding 'fhx -InputObject 'hello' -Encoding ASCII'"
+                Encoding = "ASCII"
+                Count = 1
+                Expected = "00000000   68 65 6C 6C 6F                                   hello"
+            }
+            @{
+                Name = "Can process BigEndianUnicode encoding 'fhx -InputObject 'hello' -Encoding BigEndianUnicode'"
+                Encoding = "BigEndianUnicode"
+                Count = 1
+                Expected = "00000000   00 68 00 65 00 6C 00 6C 00 6F                    .h.e.l.l.o"
+            }
+            @{
+                Name = "Can process Unicode encoding 'fhx -InputObject 'hello' -Encoding Unicode'"
+                Encoding = "Unicode"
+                Count = 1
+                Expected = "00000000   68 00 65 00 6C 00 6C 00 6F 00                    h.e.l.l.o."
+            }
+            @{
+                Name = "Can process UTF7 encoding 'fhx -InputObject 'hello' -Encoding UTF7'"
+                Encoding = "UTF7"
+                Count = 1
+                Expected = "00000000   68 65 6C 6C 6F                                   hello"
+            }
+             @{
+                Name = "Can process UTF8 encoding 'fhx -InputObject 'hello' -Encoding UTF8'"
+                Encoding = "UTF8"
+                Count = 1
+                Expected = "00000000   68 65 6C 6C 6F                                   hello"
+            }
+             @{
+                Name = "Can process UTF32 encoding 'fhx -InputObject 'hello' -Encoding UTF32'"
+                Encoding = "UTF32"
+                Count = 1
+                Expected = "00000000   68 00 00 00 65 00 00 00 6C 00 00 00 6C 00 00 00  h...e...l...l...`r`n00000010   6F 00 00 00                                      o..."
+            }
+        )
+
+    foreach ($testCase in $testCases)
+    {
+        RunEncodingTestCase $testCase
+    }
+}
 
-        $result = 65..68 | Format-Hex
-        $result | Should Not BeNullOrEmpty
-        ,$result | Should BeOfType 'Microsoft.PowerShell.Commands.ByteCollection'
-        $actualResult = $result.ToString()
-        # whitespace sensitive
-        $actualResult | should be "00000000   41 42 43 44                                      ABCD            "
+    Context "Correct Expected Errors" {
```

Comment:
Could you please rename this Context to something like "Validate Format-Hex error scenarios" or "Validate negative test cases" or "Validate error scenarios"?